app/models/physical_ticket.rb

```diff
@@ -4,4 +4,19 @@ class PhysicalTicket < ActiveRecord::Base
   has_one :conference, through: :ticket_purchase
   has_one :user, through: :ticket_purchase
   has_many :ticket_scannings
+
+  before_create :set_token
+
+  private
+
+  def set_token
+    self.token = generate_token
+  end
+
+  def generate_token
+    loop do
+      token = SecureRandom.hex(10)
+      break token unless PhysicalTicket.where(token: token).exists?
```

Comment:
I think you can also write this like:

`break token unless PhysicalTicket.exists?(token: token)`